src/Umbraco.Cms.ManagementApi/Controllers/ExamineManagement/HasIndexRebuiltExamineManagementController.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Examine;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Umbraco.Cms.ManagementApi.Factories;
+using Umbraco.Cms.ManagementApi.ViewModels.ExamineManagement;
+
+namespace Umbraco.Cms.ManagementApi.Controllers.ExamineManagement;
+
+[ApiVersion("1.0")]
+public class HasIndexRebuiltExamineManagementController : ExamineManagementControllerBase
+{
+    private readonly IExamineIndexViewModelFactory _examineIndexViewModelFactory;
+    private readonly IExamineManager _examineManager;
+
+    public HasIndexRebuiltExamineManagementController(
+        IExamineIndexViewModelFactory examineIndexViewModelFactory,
+        IExamineManager examineManager)
+    {
+        _examineIndexViewModelFactory = examineIndexViewModelFactory;
+        _examineManager = examineManager;
+    }
+
+    /// <summary>
+    ///     Check if the index has been rebuilt
+    /// </summary>
+    /// <param name="indexName"></param>
+    /// <returns></returns>
+    /// <remarks>
+    ///     This is kind of rudimentary since there's no way we can know that the index has rebuilt, we
+    ///     have a listener for the index op complete so we'll just check if that key is no longer there in the runtime cache
+    /// </remarks>
+    [HttpGet("index")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ExamineIndexViewModel), StatusCodes.Status200OK)]
+    // This endpoint for now will throw errors if the ExamineIndexViewModel ever has providerProperties defined
+    // This is because System.Text.Json cannot serialize dictionary<string, object>
+    // This has been fixed in .NET 7, so this will work when we upgrade: https://github.com/dotnet/runtime/issues/67588
```

Comment:
This comment no longer applies after removing the `ProviderProperties` from `ExamineIndexViewModel`
```suggestion
```